src/peft/peft_model.py

```diff
@@ -2227,3 +2316,255 @@ def forward(
             prompts = prompts.to(inputs_embeds.dtype)
             inputs_embeds = torch.cat((prompts, inputs_embeds), dim=1)
             return self.base_model(inputs_embeds=inputs_embeds, **kwargs)
+
+
+@dataclass
+class TunerLayerStatus:
+    name: str
+    module_type: str
+    enabled: bool
+    active_adapters: list[str]
+    merged_adapters: list[str]
+    requires_grad: dict[str, bool | Literal["irregular"]]
+    available_adapters: list[str]
+
+
+def get_layer_status(model: torch.nn.Module) -> list[TunerLayerStatus]:
```

Comment:
I would put these methods in the main init, as people can also use them for diffusers or custom models

Reply:
I was first hesitant to do that, as they were very tied to PEFT models, but with your suggestion, these functions are much more general now, so I agree and made that change.